### DISCUSSION TITLE
New Slider

### DISCUSSION THREAD

**Discussion by NirmalPuttamoni** [ant-design/ant-design#51330] [Ideas]
I have created a slider,I think this can be useful.
Can we add this slider in slider component section ?

![Screenshot 2024-10-21 204306](https://github.com/user-attachments/assets/43776cf5-d819-406e-b60b-379b1aa57ad5)
![Screenshot 2024-10-21 204259](https://github.com/user-attachments/assets/36591c95-1bca-47e7-ad1e-0b726946ddc9)

**Comment by petercat-assistant** [comment]
It seems like you have created a new slider and are suggesting it to be added to the slider component section of Ant Design. This sounds like a feature request. Unfortunately, I am unable to create new issues or feature requests directly. However, you can propose this feature by creating a new issue or discussion in the Ant Design GitHub repository, where the maintainers and community can review and consider your suggestion.

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.

**Comment by zombieJ** [comment]
What's different with current Slider component? Could you provide more detail info?

**Reply by NirmalPuttamoni** [reply]
We had a requirement in our project for a slider that starts from the middle and can be slid left and right, with the track color also starting from the middle. Although the implementation was easy, it took a lot of time for me to make a normal slider look like this. I think it would be useful for beginners like me if there were such a slider available.

**Reply by zombieJ** [reply]
Got it. Slider has the props `startPoint` but export as official API since its name is not finally decided. I guess the API can provide:

```tsx
<Slider startPoint={0} min={-100} max={100} defaultValue={0} />
```

**Comment by NirmalPuttamoni** [comment]
Hi @zombieJ ,
I am new to this. I would appreciate any feedback on my implementation and whether there are best practices I should follow.
This is how I implemented the slider : 

```import React, { useState } from 'react';
import { Slider, Switch } from 'antd';

const App: React.FC = () => {
  const [disabled, setDisabled] = useState(false);
  const [value, setValue] = useState(0);

  const startPoint = 0;
  const minValue = -100;
  const maxValue = 100;

  const onChange = (checked: boolean) => {
    setDisabled(checked);
  };
  
  const handleSliderColor = () => {
    return Number(value) === 0 ? "#69b1ff" : value > 0 ? "green" : "red";
  };

  const handleTrackStyle = () => {
    const totalRange = maxValue - minValue;
    const min = Math.min(value, startPoint);
    const max = Math.max(value, startPoint);
    return {
      left: `${((min + 100) / totalRange) * 100}%`,
      width: `${((max - min) / totalRange) * 100}%`,
      backgroundColor: handleSliderColor(),
    };
  };
  
  return (
    <>
      <Slider
        disabled={disabled}
        defaultValue={0}
        marks={{ [startPoint]: startPoint }}
        min={minValue}
        max={maxValue}
        onChange={setValue}
        styles={{ track: handleTrackStyle() }}
      />
      Disabled: <Switch size="small" checked={disabled} onChange={onChange} />
    </>
  );
};

export default App;
```
![test-slider](https://github.com/user-attachments/assets/c354aa40-4128-4509-a59b-60efa27eeea5)

**Reply by zombieJ** [reply]
As I said. Maybe we can export the `startPoint` api since it already supported:

https://stackblitz.com/edit/react-h861dq?file=demo.tsx

**Reply by NirmalPuttamoni** [reply]
Oh okay I think many people don't know that startPoint api already exists.
So we should export the **startPoint** api in SliderBaseProps interface , mention about this api in api section and create a demo slider ?

**Reply by zombieJ** [reply]
> Got it. Slider has the props startPoint but export as official API since its name is not finally decided. I guess the API can provide:

Yes. I think the name of it can be decided
